File: core/osrd-railjson/src/main/java/fr/sncf/osrd/railjson/schema/schedule/RJSPowerRestrictionRange.java
package fr.sncf.osrd.railjson.schema.schedule;

import com.squareup.moshi.Json;

public class RJSPowerRestrictionRange {
    @Json(name = "begin_position")
    public double beginPosition;

    @Json(name = "end_position")
    public double endPosition;

    @Json(name = "power_restriction_code")
    public String powerRestrictionCode;

    public RJSPowerRestrictionRange(
            double beginPosition,
            double endPosition,
            String powerRestrictionCode
    ) {
        this.beginPosition = beginPosition;
        this.endPosition = endPosition;
        this.powerRestrictionCode = powerRestrictionCode;
    }
}

File: editoast/schemas/src/paced_train/rjs_power_restriction_range.rs
use serde::Deserialize;
use serde::Serialize;
use utoipa::ToSchema;

/// A range along the train path where a power restriction is applied.
#[derive(Debug, Clone, Serialize, Deserialize, ToSchema, PartialEq)]
#[schema(example = json!({
    "begin_position": 0.0,
    "end_position": 1000.0,
    "power_restriction_code": "C1US"
}))]
pub struct RjsPowerRestrictionRange {
    /// Offset from the start of the path, in meters.
    begin_position: f32,
    /// Offset from the start of the path, in meters.
    end_position: f32,
    /// The power restriction code to apply.
    power_restriction_code: String,
}
